ecl/hthor/hthor.cpp

```diff
@@ -4929,7 +4953,7 @@ bool CHThorSelfJoinActivity::fillGroup()
     failingOuterAtmost = false;
     OwnedConstRoxieRow next;
     unsigned groupCount = 0;
-    next.setown(input->nextInGroup());
+    next.setown(groupedInput->nextInGroup());
```

Comment:
Stared at the code some more, and I am pretty sure that the limit case does not call fillGroup.